server, storage, ui: expose the lease status on the range debug page

BramGruneir opened the pull request:
Two commits:

The first converts the leaseStatus to a proto.
The second plumbs that proto to the ui.

I'm going to try to try to get the liveness of all the replicas next.

BramGruneir committed:
stroage: convert leaseStatus to a proto

This will facilitate exposing the lease status to a status endpoint and the ui.

BramGruneir committed:
server, storage, ui: expose the lease state on the range debug page

Closes #18339.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18707)
<!-- Reviewable:end -->

BramGruneir commented:
Almost forgot the screenshots:
<img width="902" alt="screen shot 2017-09-22 at 22 19 35" src="https://user-images.githubusercontent.com/1614265/30769383-254af3e2-9fe5-11e7-98a7-359f92032bd0.png">
<img width="870" alt="screen shot 2017-09-22 at 22 19 22" src="https://user-images.githubusercontent.com/1614265/30769384-254fce44-9fe5-11e7-9133-a945bdf3524d.png">

mrtracy commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 6 files at r1.
Review status: 2 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/storage/lease_status.proto, line 34 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707#-KuuRYhUHB5UxIgaVfLo:-KuuRYhUHB5UxIgaVfLp:bw8ja2l) ([raw file](https://github.com/cockroachdb/cockroach/blob/fda658649e996eeb86a0a9af5815f54eac60a70d/pkg/storage/lease_status.proto#L34)):*
> ```protobuf
>   // EXPIRED indicates that the lease can't be used.
>   EXPIRED = 3;
>   // Proscribed indicates that the lease's proposed timestamp is earlier than
> ```

You didn't capitalize the enum value in just this one comment... Why?! It's all I can think about!!

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 196 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707#-KuuSFmFI5FL7EwDMPV0:-KuuSFmFI5FL7EwDMPV1:b-ndf96o) ([raw file](https://github.com/cockroachdb/cockroach/blob/fda658649e996eeb86a0a9af5815f54eac60a70d/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L196)):*
> ```XML
>       classNames.push("range-table__cell--dormant");
>     } else {
>       classNames = _.concat(classNames, cell.className);
> ```

Why use concat here, but push everywhere else in this function?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707#-:-KuuSheL2VORHWx2cdDY:bnfp4nl)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 1 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/storage/lease_status.proto, line 34 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707#-KuuRYhUHB5UxIgaVfLo:-KuygPf5CiEtWNe7YGUi:bvx5xp) ([raw file](https://github.com/cockroachdb/cockroach/blob/fda658649e996eeb86a0a9af5815f54eac60a70d/pkg/storage/lease_status.proto#L34)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

You didn't capitalize the enum value in just this one comment... Why?! It's all I can think about!!
</blockquote></details>

Fixed!  You can now rest easy.

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 196 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707#-KuuSFmFI5FL7EwDMPV0:-Kuygb7bLZcapV8PuawN:b-etjqj8) ([raw file](https://github.com/cockroachdb/cockroach/blob/fda658649e996eeb86a0a9af5815f54eac60a70d/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L196)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

Why use concat here, but push everywhere else in this function?
</blockquote></details>

Because cell.className is an array of strings.  
Changed it to a push with the spread operator `...`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18707)*
<!-- Sent from Reviewable.io -->
